### PR TITLE
WES pipeline conf generation

### PR DESCRIPTION
**Pull request by curlup**
this will be accompanied with an API PR that will call that and send configs as email attachments

### CODE REVIEW

File: cidc_schemas/prism.py [hunk 1]
```diff
@@ -974,9 +1075,10 @@ def get_schema(self, walk, schema, **kwargs):
 
 PRISM_MERGE_STRATEGIES = {
     # This overwrites the default jsonmerge merge strategy for literal values.
-    "overwrite": ThrowOnCollision(),
+    # "overwrite": ThrowOnCollision(),
```
Comment (curlup) [inline cidc_schemas/prism.py]: woops

File: cidc_schemas/wes_analysis_config.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Samples: a map of sample name to a list of Google bucket paths to input files, 
+# VALID INPUTS: fastq, fastq.gz
+# NOTE: for PAIRED-END fastq/fastq.gz, give both pairs to the sample
+samples:
+  # normal
+  {{normal_sample.cimac_id}}:
+{% if normal_sample.files.r1 %}
+    # r1  
+  {% for fastq_gz_artifact in normal_sample.files.r1 %}
+    - gs://{{BIOFX_BUCKET_NAME}}/{{fastq_gz_artifact.object_url}}
```
Comment (jacoblurye) [inline cidc_schemas/wes_analysis_config.yaml.j2]: Where does the value for `BIOFX_BUCKET_NAME` come from? I don't see it being passed when this template is rendered.

File: cidc_schemas/wes_analysis_config.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  {% endfor %}
+{% else %}
+    # bam
+  {% for bam_artifact in tumor_sample.files.bam %}
+    - gs://{{BIOFX_BUCKET_NAME}}/{{bam_artifact.object_url}}
+  {% endfor %}
+{% endif %}
+  
+# metahseet- Group the samples into Tumor/Normal "runs"
```
Comment (jacoblurye) [inline cidc_schemas/wes_analysis_config.yaml.j2]: "metasheet"

File: cidc_schemas/wes_analysis_config.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Samples: a map of sample name to a list of Google bucket paths to input files,
```
Comment (jacoblurye) [inline cidc_schemas/wes_analysis_config.yaml.j2]: I'm not sure at the top-level in the `cidc_schemas` directory is the best place for this jinja file to live - maybe a subdirectory?

Also, will this file definitely be bundled with the package when it's deployed?

File: tests/test_prism.py [hunk 5]
```diff
@@ -919,6 +928,9 @@ def test_end_to_end_prismify_merge_artifact_merge(xlsx, template):
                 == 1
             )
 
+            with pytest.raises(Exception, match="been here"):
```
[the remark this answers is not in the document]
Comment (curlup) [reply tests/test_prism.py]: It's just me being tired on Friday I guess.
Thanks for spotting.

File: tests/test_prism.py [hunk 6]
```diff
@@ -43,7 +47,12 @@
 from cidc_schemas.template_reader import XlTemplateReader
 
 from .constants import ROOT_DIR, SCHEMA_DIR, TEMPLATE_EXAMPLES_DIR, TEST_DATA_DIR
-from .test_templates import template_set
+from .test_templates import (
+    template_set,
+    template,
```
[the remark this answers is not in the document]
Comment (curlup) [reply tests/test_prism.py]: It doesn't work without it somehow

File: tests/test_prism.py [hunk 7]
```diff
@@ -1631,6 +1643,112 @@ def test_prism_many_artifacts_from_process_as_on_one_record(monkeypatch):
...
+
+
+@pytest.fixture(scope="session")
+def prism_patch(prismify_result):
+    prism_patch, _, _ = prismify_result
+    return prism_patch
+
+
+@pytest.fixture(scope="session")
+def prism_fmap(prismify_result):
```
[the remark this answers is not in the document]
Comment (curlup) [reply tests/test_prism.py]: removed this unused fixture

File: tests/test_prism.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jacoblurye) [reply tests/test_prism.py]: Interesting...no big deal either way!

File: tests/test_prism.py [hunk 8]
```diff
@@ -1631,6 +1639,110 @@ def test_prism_many_artifacts_from_process_as_on_one_record(monkeypatch):
...
+
+    patch_with_artifacts = prism_patch_stage_artifacts(prismify_result, template.type)
+
+    # if it's an analysis - we need to merge corresponding preliminary assay first
+    prelim_assay = stage_assay_for_analysis(template.type)
+    if prelim_assay:
+        full_ct, errs = merge_clinical_trial_metadata(prelim_assay, full_ct)
+        assert 0 == len(errs)
+
+    full_ct, errs = merge_clinical_trial_metadata(patch_with_artifacts, full_ct)
```
Comment (jacoblurye) [inline tests/test_prism.py]: Is it necessary to merge `patch_with_artifacts` and `full_ct` before running `generate_analysis_configs_from_upload_patch`? My impression is that it isn't, but just want to check my understanding.

File: cidc_schemas/prism.py [hunk 9]
```diff
@@ -918,6 +929,152 @@ def merge_artifact_extra_metadata(
...
+    templ = jinja2.Environment(loader=loader).get_template(
+        "pipeline_configs/wes_analysis_config.yaml.j2"
+    )
+
+    class AnalysisRun(NamedTuple):
+        normal_cimac_id: str
+        tumor_cimac_id: str
+        run_id: str
+
+    def internal(full_ct: dict, patch: dict, bucket: str) -> Dict[str, str]:
```
Comment (jacoblurye) [inline cidc_schemas/prism.py]: Should `full_ct` have `patch` merged in at this point? If so or if not, it'd be helpful to have a note about that here.

**Pull request merged**